akka-distributed-data/src/main/scala/akka/cluster/ddata/ReplicatedData.scala

```diff
@@ -37,6 +37,17 @@ trait ReplicatedData {
 
 }
 
+trait DeltaReplicatedData extends ReplicatedData {
+  type D <: ReplicatedData
+
+  def delta: Option[D]
+
+  def mergeDelta(d: D): T
+
```

Comment:
In line with the above, I would also consider dropping mergeDelta. There is no functional difference between merging a delta and merging a full state. Any performance gains one might foresee with a delta-specific implementation could also be nullified once the delta is nearly the size of the full state. If you'd choose to keep it, for just the potential of gaining performance, I would suggest letting it have a default implementation of merge.